### ISSUE TITLE
oauth2orize.exchange with HTTP Basic scheme

### ISSUE THREAD

**Issue by AntonioFernandezFernandez** [jaredhanson/oauth2orize#122]
oauth2orize.exchange.password requires username and password request params in url, but if i want to protect the `token` endpoint with basic http authentication (which is more secure than ClientPasswordStrategy), there is no way to exchange encrypted username:password key pair in header for access token. Any suggestions?

**Comment by AntonioFernandezFernandez** [comment]
any news????

**Comment by AntonioFernandezFernandez** [comment]
please! any news???? any people working on this bug?

**Comment by jaredhanson** [comment]
I think you misunderstand what's going on here.

The `basic` or `oauth2-client-password` authentication will authenticate the *client* making the `/token` request.

The `username` and `password` are the *user* credentials.  Those need to be verified in the exchange prior to issuing a user-specific access token.

**Comment by efkan** [comment]
Hi @AntonioFernandezFernandez,

I've been trying to use `Resource Owner Password Credentials` too.

The [specification](https://tools.ietf.org/id/draft-ietf-oauth-v2-27.html#rfc.section.4.3)  says:

> the client MUST authenticate with the authorization server as described in [Section 3.2.1](https://tools.ietf.org/id/draft-ietf-oauth-v2-27.html#token-endpoint-auth).

Actually the latest paragraph of the section 3.2.1 says 

> A public client that was not issued a client password MAY use the client_id request parameter to identify itself when sending requests to the token endpoint (e.g. for the purpose of providing end-user context, client usage statistics).

However, it just may.

I'm thinking give a standart `client_id` like `AwesomeAPP` and a `client_secret` like `Aa1BBb_*0`.
Or I may use  `AwesomeAPP_v1.0`  as `client_id` to track that how many people use which version of the App. I know it's not the right way but I can.

In short `client_id` and `client_secret` are used for the authorization of the **`APP`**.
`username` and `password` are used for the authorization of the **`user`**.

Nevertheless, confirming or fixing the all of these by @jaredhanson would be great!
